first_script.py

```diff
@@ -522,4 +522,81 @@ def make_spider (row, title, color):
 plt.savefig(mypath +'corticothalamic_FSL.png', dpi=None, facecolor='w', edgecolor='w',
         orientation='landscape', papertype=None, format=None,
         transparent=False, bbox_inches=None, pad_inches=0.1,
-        frameon=None, metadata=None)
\ No newline at end of file
+        frameon=None, metadata=None)
+
+##############################################################################
+# predict presence of verbal perseveration by applying random forrest analysis
+
+#verbal perseverations early postop
+
+X=ANTS_ratio.iloc[:, 0:46]
+Y=behavior.iloc[:,3]
+
+#remove subject with nan value from both datasets (here the second line)
+X=X.drop(X.index[1])
+Y=Y.drop(Y.index[1])
+
+from sklearn.model_selection import train_test_split
+X_train, X_test, Y_train, Y_test = train_test_split (X, Y, test_size=0.2, random_state=0)
+
+#this normalisation step makes the performance drop
+#from sklearn.preprocessing import StandardScaler
+#sc = StandardScaler ()
+#X_train= sc.fit_transform (X_train)
+#X_test = sc.fit_transform (X_test)
+
+from sklearn.ensemble import RandomForestClassifier
+classifier=RandomForestClassifier (n_estimators=45, random_state=0)
+classifier.fit(X_train, Y_train)
+Y_pred= classifier.predict(X_test)
+
+from sklearn.metrics import classification_report, confusion_matrix, accuracy_score
+print (confusion_matrix (Y_test,Y_pred))
+print (classification_report(Y_test,Y_pred))
+print (accuracy_score(Y_test,Y_pred))
+
+importances = classifier.feature_importances_
+std = np.std([tree.feature_importances_ for tree in classifier.estimators_],
+             axis=0)
+indices = np.argsort(importances)[::-1]
+X_column_values=X.columns.values
+
+# Print the feature ranking
+print("Feature ranking:")
+
+for f in range(X.shape[1]):
+    print("%d. feature %s (%f)" % (f + 1, X_column_values[indices[f]], importances[indices[f]]))
+
+# Plot the impurity-based feature importances of the forest
+plt.figure()
+plt.title("Feature importances")
+plt.bar(range(X.shape[1]), importances[indices],
+        color="r", yerr=std[indices], align="center")
+plt.xticks(range(X.shape[1]), X_column_values[indices],rotation=90,size=7)
+plt.xlim([-1, X.shape[1]])
+plt.show()
+
+#afficher les réseaux 4 et 12 de Yeo dans NiLearn
```

Comment:
why have you commented the following lines ?

Reply:
I wanted to display Yeo 4th and 12th network since they seemed to be important for the classifier. It didn't work and I was planning on fixing it later. My second thought is that plotting Yeo's networks is not very helpful at this stage.